File: Parsing/Score of Parentheses/stack.ts
// https://leetcode.com/problems/score-of-parentheses/
function scoreOfParentheses(S: string): number {
    let lst=[];
    for (let i =0; i< S.length; i++){
        if (S[i] === ')'){
            let sum=0;
            while(lst.length>0 && lst[lst.length-1] !== '('){
                sum+=lst.pop();
            }
            lst.pop();
            if (sum==0) lst.push(1);
            else lst.push(2*sum);
        }
        else{
            lst.push('(');
        }
        // console.log(lst);
    }
    return lst.reduce((a, b) => a + b, 0)
};

File: Trees&Tries/(E) N-ary Tree Preorder Traversal/stack.py
# https://leetcode.com/problems/n-ary-tree-preorder-traversal/
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def preorder(self, root: 'Node') -> List[int]:
        lst=[]
        stack=[]
        curr=root
        while(curr or len(stack)>0):
            while(curr):
                lst.append(curr.val)
                stack.append(curr)
                if len(curr.children)>0:
                    curr=curr.children.pop(0)
                else: curr=None
            
            curr=stack.pop()
            if len(curr.children)>0:
                stack.append(curr)
                curr=curr.children.pop(0)
            else: curr=None
        return lst
                    